core/docker_service.go

```diff
@@ -326,14 +340,37 @@ func (ds *dockerService) AlphaVersion(
 }
 
 // getDockerVersion gets the version information from docker.
-func (ds *dockerService) getDockerVersion() (*dockertypes.Version, error) {
-	v, err := ds.client.Version()
+func (ds *dockerService) getDockerVersion() (v *dockertypes.Version, err error) {
+	if ds.versionCache != nil {
+		v, err = ds.getDockerVersionFromCache()
+	} else {
+		v, err = ds.client.Version()
```

Comment:
I think you would want to add the version to cache here since it was a miss and you had to look it up from the client

Reply:
The modified APIversion is already cached.